src/main/resources/liquibase/crop_changelog/21_3_0.xml

```diff
@@ -0,0 +1,814 @@
+				<sqlCheck expectedResult="0">
+					SELECT count(1) FROM stockprop WHERE cvalue_id IS NOT NULL;
+				</sqlCheck>
+				<sqlCheck expectedResult="1">
+					SELECT count(*) > 0
+					FROM (SELECT DISTINCT( sp.type_id ) AS typeId
+						  FROM stockprop sp
+								   INNER JOIN cvterm_relationship cr
+						  ON sp.type_id = cr.subject_id
+							  AND cr.type_id = 1220
+								   INNER JOIN cvterm_relationship cr2
+						  ON cr.object_id = cr2.subject_id
+							  AND cr2.type_id = 1190) AS inn;
+				</sqlCheck>
+			</and>
+		</preConditions>
+		<comment>Migrate categorical variable values to 'cvalue_id' column</comment>
+		<sql dbms="mysql" splitStatements="true">
+			UPDATE stockprop sp
+			SET cvalue_id = value,
+				value     = (SELECT name
+							 FROM cvterm
+							 WHERE cvterm_id = cvalue_id)
+			WHERE value IN (SELECT cvterm_id
+							FROM cvterm_relationship cr
+									 INNER JOIN cvterm_relationship cr2 ON cr2.subject_id = cr.object_id AND cr2.type_id = 1190
+									 INNER JOIN cvterm c ON c.cvterm_id = cr2.object_id
+							WHERE cr.subject_id = sp.type_id AND cr.type_id = 1220) AND
+					type_id IN (SELECT inn.typeid
+								FROM (SELECT DISTINCT(sp.type_id) AS typeId
+									  FROM stockprop sp
+											   INNER JOIN cvterm_relationship cr ON sp.type_id = cr.subject_id AND cr.type_id = 1220
+											   INNER JOIN cvterm_relationship cr2 ON cr.object_id = cr2.subject_id AND cr2.type_id = 1190) AS inn);
+		</sql>
+	</changeSet>
+	<!-- End of ENTRY_TYPE related changesets -->
+
+	<changeSet author="lkovacic" id="v21.3.0-24">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT COUNT(*) > 0
+				FROM stockprop sp
+					 INNER JOIN cvtermprop cp ON cp.cvterm_id = sp.type_id AND cp.type_id = 1800
+				WHERE cp.value = 'Germplasm Descriptor';
+			</sqlCheck>
+		</preConditions>
+		<comment>Migrate remaining germplasm descriptors to entry details in ontology and 'projectprop' table</comment>
+		<sql dbms="mysql" splitStatements="true">
+			INSERT INTO entry_detail_migration_output (cvterm_id)
+			SELECT DISTINCT sp.type_id
+								 FROM stockprop sp
+										  INNER JOIN cvtermprop cp
+								 ON cp.cvterm_id = sp.type_id
+									 AND cp.type_id = 1800
+								 WHERE cp.value = 'Germplasm Descriptor';
+
+			UPDATE cvtermprop cp
+				INNER JOIN stockprop sp ON sp.type_id = cp.cvterm_id AND cp.type_id = 1800
+			SET cp.value = 'Entry Detail'
+			WHERE cp.value = 'Germplasm Descriptor';
+
+			UPDATE projectprop
+			SET type_id = 1815
+			WHERE type_id = 1804 AND
+				variable_id IN (SELECT DISTINCT sp.type_id
+								 FROM stockprop sp
+										  INNER JOIN cvtermprop cp
+								 ON cp.cvterm_id = sp.type_id
+									 AND cp.type_id = 1800
+								 WHERE cp.value = 'Germplasm Descriptor');
+		</sql>
+	</changeSet>
+
+	<!-- Identify unknown categorical values changesets -->
+	<changeSet author="lkovacic" id="v21.3.0-25">
+		<preConditions onFail="MARK_RAN">
+		  <not>
```

Comment:
@lik-83 I would modify this precondition to create this table only when there are invalid values to move